src/olympia/api/middleware.py

```diff
@@ -13,7 +25,7 @@ class GZipMiddlewareForAPIOnly(GZipMiddleware):
     """
 
     def process_response(self, request, response):
-        if not request.path.startswith('/api/'):
```

Comment:
The effect of testing for `startswith('/api/')` here was legacy API requests weren't handled as API requests so didn't have the GZipMiddleware enabled (because the `.path` before the locale middleware gets invoked starts with `/en-US/firefox/` (or whatever).  

I'm not sure it that was deliberate, but the behaviour will be the same after this patch.  (It doesn't matter that much either way as we'll hopefully be dropping the legacy API before the end of the quarter)

Reply:
Yeah, I think it was deliberate to not cause any legacy issues since we didn't know how Firefox would handle Gzip in the API. So it's good we're not changing this.